src/tls_wrap.cc

```diff
@@ -1076,6 +1079,131 @@ int TLSWrap::SelectSNIContextCallback(SSL* s, int* ad, void* arg) {
   return SSL_TLSEXT_ERR_OK;
 }
 
+#ifndef OPENSSL_NO_PSK
+
+void TLSWrap::SetPskIdentityHint(const FunctionCallbackInfo<Value>& args) {
+  TLSWrap* p;
+  ASSIGN_OR_RETURN_UNWRAP(&p, args.Holder());
+  CHECK_NOT_NULL(p->ssl_);
+
+  Environment* env = p->env();
+
+  CHECK(args[0]->IsString());
+  node::Utf8Value hint(env->isolate(), args[0].As<String>());
+
+  if (!SSL_use_psk_identity_hint(p->ssl_.get(), *hint)) {
+    Local<Value> err = Exception::Error(env->psk_identity_hint_error());
```

Comment:
Can we use an error in the style of `node_errors.h` for this?

Reply:
Would look much better, kind of missed such a possibility. Done.